### DISCUSSION TITLE
Couldn't make a post on r/ChatGPT on Reddit

### DISCUSSION THREAD

**Discussion by ykdojo** [ykdojo/kaguya#36] [General]
I tried using different flairs, with or without a link, shorter and longer, but nothing I tried seems to work.

Not sure why - does anyone have any idea?

Here's (one of) the full title & text I tried:

---

Title: I gave ChatGPT access to my file system, and it's been really interesting

Body:

Hi, long time lurker, first time poster here.

I've been experimenting with this idea of giving ChatGPT access to my file system for a while, and it's been really fascinating, so I wanted to share my experience with you all here.

---

tldr: I gave ChatGPT access to my file system and it's been awesome

---

First, you might ask, how?

To do this, I built a ChatGPT plugin for it. Luckily I was able to get access to developing plugins just by getting on their waitlist [here](https://openai.com/waitlist/plugins).

Then I thought, I don't want to give ChatGPT access to my entire file system. So I used Docker for it and containerized it so it only has access to files that you add to the project directory. (Note, if you're not technical, don't worry too much about that sentence.)

Anyway, with that basic structure, I got to building it -

I remember I first built the ability for it to make a file, then ran it - and it worked! Pretty cool.

Then I added the ability for it to read files, edit files, rename them, etc. - the whole suite. It actually built much of it itself with my instruction! It did make mistakes from time to time, but with its coach (me) being there, it was able to make pretty good progress improving itself.

Then I added the ability for it to be able to execute code.

Now again, since the whole plugin/application is containerized, it's only able to do stuff in this sandboxed environment.

Anyway, I remember telling it to commit changes it's made, and it did it! I then told it to do some complex Git operations, and it was able to do those, too. Note for non-techies - Git is just a way to control different versions of your code.

Then I recorded a demo video of how I used it, put it in the same folder as this project, and asked it to edit the video with a Python script - and it did it pretty much flawlessly, also.

---

So since then, I've been using it to mostly improve itself but also to work on my personal projects, and it's been awesome!

It's mostly for programmers since it does require some amount of technical knowledge to use it, but I've used it to write a short novel, do quick video editing, write and execute website scraping scripts, etc. - it's been really fun and I think it does boost productivity a lot, even compared to just using ChatGPT normally.

---

Okay I wrote a lot, but I just wanted to share this experience with you all here since I know you guys are really passionate about ChatGPT in this sub.

In case you want to check out the source code of this plugin, it's right here: https://github.com/ykdojo/kaguya/tree/master (Mods - I'm not trying to self-promote here, so I hope it's okay. If not, please let me know and I'll remove the link.)

**Comment by ThirDecade2020** [comment]
This is a good read. Any luck posting on Reddit yet? Besides your social media, Kaguya has only been shared on Hacker News? A nice media campaign might be a good project for this repo in future.

**Reply by ykdojo** [reply]
Yes pretty much. And no I tried posting on Reddit a few times after that, but it never worked for me

**Reply by ThirDecade2020** [reply]
That's weird, they just don't allow that type of post? Are they afraid of AI? Ramping up exposure for Kaguya is a great idea and there are so many ways to go about it now with AI at the forefront of everything. I have been looking at a variety of AI tools and there is a lot of useful stuff out there, so best bet might be to start that process so Kaguya can snatch its piece before more AI comes flooding in - if that's your vision of course. I would love to work on this I have a strong business background looking at startups entering the market and I am also a contributor here gaining knowledge of code base. Curious about your grand vision for Kaguya!

**Comment by Charlotte-br560** [comment]
Your experimentation with ChatGPT and Docker for file system access is impressive! Regarding the Reddit post issue, it might be due to subreddit filters. Have you tried reaching out to mods for help? Also, sharing your plugin's source code is great. Keep up the good work!
